libraries/AP_HAL/QSPIDevice.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Code by Andrew Tridgell and Siddharth Bharat Purohit
+ */
+#pragma once
+
+#include <inttypes.h>
+#include <vector>
+
+#include "AP_HAL_Namespace.h"
+#include "Device.h"
+#include "utility/OwnPtr.h"
+
+#ifndef HAL_USE_QSPI_DEFAULT_CFG
+#define HAL_USE_QSPI_DEFAULT_CFG 1
+#endif
+
+namespace AP_HAL
+{
+
+// Underlying HAL implementation can override these
+#if HAL_USE_QSPI_DEFAULT_CFG
+namespace QSPI
+{
+constexpr uint32_t CFG_CMD_MODE_MASK        =   (3LU << 8LU);
```

Comment:
This is redefing stuff that is already defined in chibios - why is that necessary?

Reply:
So, the reason is we do not want to create dependency of Drivers running on QSPI on ChibiOS. There might be other HALs that might want to use those drivers.